ts_scripts/torchserve_grpc_client.py

```diff
@@ -64,25 +65,56 @@ def register(stub, model_name, mar_set_str):
 def unregister(stub, model_name):
     try:
         response = stub.UnregisterModel(
-            management_pb2.UnregisterModelRequest(model_name=model_name))
+            management_pb2.UnregisterModelRequest(model_name=model_name)
+        )
         print(f"Model {model_name} unregistered successfully")
     except grpc.RpcError as e:
         print(f"Failed to unregister model {model_name}.")
         print(str(e.details()))
         exit(1)
 
 
-if __name__ == '__main__':
-    # args:
-    # 1-> api name [infer, register, unregister]
-    # 2-> model name
-    # 3-> model input for prediction
-    args = sys.argv[1:]
-    if args[0] == "infer":
-        infer(get_inference_stub(), args[1], args[2])
-    else:
-        api = globals()[args[0]]
-        if args[0] == "register":
-            api(get_management_stub(), args[1], args[2])
-        else:
-            api(get_management_stub(), args[1])
+if __name__ == "__main__":
+
+    parent_parser = argparse.ArgumentParser(add_help=False)
+    parent_parser.add_argument(
+        "model_name",
+        type=str,
+        default=None,
+        help="Name of the model used.",
+    )
+
+    parser = argparse.ArgumentParser(
+        description="TorchServe gRPC client",
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+    subparsers = parser.add_subparsers(help="Action", dest="action")
+
+    A_parser = subparsers.add_parser("infer", parents=[parent_parser], add_help=False)
```

Comment:
Nitpick: In place of the variable name `A_parser`, could we use a name that indicates what the variable holds, say for example `infer_action_parser` or something similar? Same for `B_parser` and `C_parser` as well. I think this would help make the code easier to read and maintain.

Reply:
Yeah I agee with that. Thanks :)